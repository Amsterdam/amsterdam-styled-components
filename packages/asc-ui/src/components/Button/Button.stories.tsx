import styled from '@datapunt/asc-core'
import {
  Download,
  ChevronRight,
  ChevronLeft,
  Close,
  Facebook,
  Twitter,
  Linkedin,
  Email,
  Enlarge,
  Print,
} from '@datapunt/asc-assets'
import { storiesOf } from '@storybook/react'
import * as React from 'react'
import Button from './Button'
import ShareButton from './ShareButton'

const ButtonBar = styled.div<{}>`
  display: flex;
  height: 100px;

  & > * {
    margin-right: 5px;
  }
`

storiesOf('Atoms/Buttons', module)
  .addDecorator(storyFn => (
    <div style={{ padding: '40px 10px' }}>{storyFn()}</div>
  ))
  .add('[deprecated] variants', () => (
    <ButtonBar>
      <Button>Default button</Button>
      <Button color="primary">Primary button</Button>
      <Button color="secondary">Secondary button</Button>
    </ButtonBar>
  ))
  .add('variants', () => (
    <ButtonBar>
      <Button variant="primary">Default button</Button>
      <Button variant="application" iconLeft={<Enlarge />} iconSize={10}>
        Application button
      </Button>
      <Button
        variant="secondary"
        iconLeft={<Download />}
        iconRight={<ChevronRight />}
      >
        Download button
      </Button>
      <Button variant="secondary" taskflow>
        Taskflow button
      </Button>
      <Button variant="tertiary" iconRight={<ChevronRight />}>
        Download button
      </Button>
      <Button variant="primaryInverted" iconLeft={<ChevronRight />}>
        Download button
      </Button>
      <Button variant="textButton" iconLeft={<ChevronLeft />} iconSize={14}>
        Text button
      </Button>
    </ButtonBar>
  ))
  .add('variants - disabled', () => (
    <ButtonBar>
      <Button disabled variant="primary">
        Default button
      </Button>
      <Button
        disabled
        variant="secondary"
        iconLeft={<Download />}
        iconRight={<ChevronRight />}
      >
        Download button
      </Button>
      <Button disabled variant="tertiary" iconRight={<ChevronRight />}>
        Download button
      </Button>
      <Button disabled variant="primaryInverted" iconLeft={<ChevronRight />}>
        Download button
      </Button>
      <Button disabled variant="secondary" taskflow>
        Taskflow button
      </Button>
      <Button
        disabled
        variant="textButton"
        iconLeft={<ChevronLeft />}
        iconSize={14}
      >
        Text button
      </Button>
    </ButtonBar>
  ))
  .add('with only an icon', () => (
    <>
      <Button size={30} variant="blank" icon={<ChevronRight />} />
      <Button size={60} variant="blank" iconSize={40} icon={<Close />} />
    </>
  ))
  .add('social buttons', () => (
    <ButtonBar>
      <ShareButton hoverColor="#3b5999" iconSize={30}>
        <Facebook />
      </ShareButton>
      <ShareButton hoverColor="#55acee">
        <Twitter />
      </ShareButton>
      <ShareButton hoverColor="#0077B5">
        <Linkedin />
      </ShareButton>
      <ShareButton>
        <Email />
      </ShareButton>
      <ShareButton>
        <Print />
      </ShareButton>
    </ButtonBar>
  ))
