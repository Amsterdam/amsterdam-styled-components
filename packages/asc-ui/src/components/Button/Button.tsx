import * as React from 'react'
import ButtonStyle, {
  IconLeft,
  IconRight,
  Props as ButtonStyleProps,
  ButtonVariants,
} from './ButtonStyle'
import Icon, { defaultProps as iconDefaultProps } from '../Icon/Icon'

export type Props = {
  as?: keyof JSX.IntrinsicElements | React.ComponentType<any>
  iconLeft?: React.ReactNode
  iconRight?: React.ReactNode
  icon?: React.ReactNode
  iconSize?: number
  $as?: any
} & ButtonStyleProps &
  React.ButtonHTMLAttributes<HTMLButtonElement>

export { ButtonVariants }

const Button: React.FC<Props> = ({
  children,
  iconLeft,
  iconRight,
  icon,
  iconSize,
  $as,
  ...otherProps
}): any => {
  const iconProps = {
    size: iconSize || iconDefaultProps.size,
  }
  return (
    <ButtonStyle {...otherProps} as={$as}>
      {iconLeft && <IconLeft {...iconProps}>{iconLeft}</IconLeft>}
      {icon ? <Icon {...iconProps}>{icon}</Icon> : children}
      {iconRight && <IconRight {...iconProps}>{iconRight}</IconRight>}
    </ButtonStyle>
  )
}

export default Button
