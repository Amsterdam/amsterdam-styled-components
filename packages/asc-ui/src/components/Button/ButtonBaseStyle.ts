import { transitions } from 'polished'
import styled, { Theme } from '@datapunt/asc-core'
import { color as themeColor, focusStyleOutline } from '../../utils'
import { flexboxMinHeightFix } from '../shared/ie-fixes'

export type Props = {
  color?: Theme.TypeLevel
  square?: boolean
  hoverColor?: string
  href?: string
  size?: number
}

const ButtonBaseStyle = styled.button<Props>`
  display: inline-flex;
  align-items: center;
  text-decoration: none;
  border: none;
  cursor: pointer;
  font-size: 16px;
  font-weight: normal;
  padding: 0 10px 0 10px;
  ${focusStyleOutline()}
  ${transitions(['color', 'background-color'], '0.1s ease-in-out')}
  background: ${({ color, theme }) => themeColor(color)({ theme })};

  &:hover {
    background: ${({ color, theme }) =>
      color
        ? themeColor(color, 'dark')({ theme })
        : themeColor('tint', 'level3')({ theme })};
  }

  ${flexboxMinHeightFix()}
`

export default ButtonBaseStyle
