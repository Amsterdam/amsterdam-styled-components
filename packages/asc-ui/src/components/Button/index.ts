export { ButtonProps, default } from './Button'
