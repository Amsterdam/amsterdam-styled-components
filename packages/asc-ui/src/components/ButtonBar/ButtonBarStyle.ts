import styled from '@datapunt/asc-core'

export type Props = {
  padding?: number
}

const ButtonBarStyle = styled.div<Props>`
  display: flex;
  ${({ padding }) => `padding: ${padding}px;`}

  & > * {
    margin-right: 5px;
  }
`

ButtonBarStyle.defaultProps = {
  padding: 5,
}

export default ButtonBarStyle
