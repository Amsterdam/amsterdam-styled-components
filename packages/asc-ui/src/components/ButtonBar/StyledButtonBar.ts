import styled from 'styled-components'

export type Props = {
  padding?: number
}

const StyledButtonBar = styled.div<Props>`
  display: flex;
  ${({ padding }) => `padding: ${padding}px;`}

  & > * {
    margin-right: 5px;
  }
`

StyledButtonBar.defaultProps = {
  padding: 5,
}

export default StyledButtonBar
