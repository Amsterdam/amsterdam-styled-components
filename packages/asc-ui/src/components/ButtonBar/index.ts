export { default, Props } from './ButtonBar'
