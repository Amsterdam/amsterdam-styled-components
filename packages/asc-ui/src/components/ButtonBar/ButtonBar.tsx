import React from 'react'
import ButtonBarStyle, {
  ButtonBarStyleProps,
} from '../../styles/components/ButtonBarStyle'

export type Props = {
  styledComponent?: any
} & ButtonBarStyleProps

const ButtonBar: React.FC<Props> = ({ children, styledComponent }) => {
  const ExtendedButtonBarStyle = styledComponent
  return (
  <ExtendedButtonBarStyle padding={0}>{children}</ExtendedButtonBarStyle>
)}

ButtonBar.defaultProps = {
  styledComponent: ButtonBarStyle
}

export default ButtonBar
export {ButtonBarStyle}
