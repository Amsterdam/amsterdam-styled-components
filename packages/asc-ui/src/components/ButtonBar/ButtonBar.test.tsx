import * as React from 'react'
import { shallow } from 'enzyme'
import Button from '../Button'
import ButtonBar from './ButtonBar'
import 'jest-styled-components'

describe('ButtonBar', () => {
  it('should render the buttons in the bar', () => {
    const component = shallow(
      <ButtonBar>
        <Button variant="primary">Hello</Button>
        <Button variant="secondary">World</Button>
      </ButtonBar>,
    ).dive()
    expect(component).toMatchSnapshot()
  })
})
