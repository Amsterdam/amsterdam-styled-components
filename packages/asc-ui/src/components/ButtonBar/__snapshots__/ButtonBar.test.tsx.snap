// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonBar should render the buttons in the bar 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "_foldedDefaultProps": Object {
        "padding": 5,
      },
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  ",
          [Function],
          "

  & > * {
    margin-right: 5px;
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  padding={0}
>
  <Button
    color="primary"
  >
    Hello
  </Button>
  <Button
    color="secondary"
  >
    World
  </Button>
</StyledComponent>
`;
